mm2src/mm2_main/tests/docker_tests/eth_docker_tests.rs

```diff
@@ -1166,3 +1046,350 @@ fn test_nonce_lock() {
+        Some(amount),
+        ContractType::Erc1155,
+        geth_nft_maker_swap_v2(),
+        geth_erc1155_contract(),
+        time_lock_to_refund,
+    );
+
+    let maker_address = block_on(setup.maker_global_nft.my_addr());
+    let balance = geth_erc1155_balance(maker_address, U256::from(token_id));
+    assert_eq!(U256::from(amount), balance);
+
+    let swap_amount = 2u32;
+    let maker_payment_to_refund = send_nft_maker_payment(&setup, swap_amount.into());
+    log!(
+        "Maker sent ERC1155 NFT payment, tx hash: {:02x}",
+        maker_payment_to_refund.tx_hash()
+    );
+
+    wait_for_confirmations(&setup.maker_global_nft, &maker_payment_to_refund, 150);
+
+    let swap_contract_balance = geth_erc1155_balance(geth_nft_maker_swap_v2(), U256::from(token_id));
+    assert_eq!(U256::from(swap_amount), swap_contract_balance);
+    let balance = geth_erc1155_balance(maker_address, U256::from(token_id));
+    assert_eq!(U256::from(1u32), balance);
+
+    let refund_timelock_tx = refund_nft_maker_payment(
+        &setup,
+        &maker_payment_to_refund,
+        &ContractType::Erc1155,
+        RefundType::Timelock,
+    );
+    log!(
+        "Maker refunded ERC1155 NFT payment after timelock, tx hash: {:02x}",
+        refund_timelock_tx.tx_hash()
+    );
+
+    wait_for_confirmations(&setup.maker_global_nft, &refund_timelock_tx, 150);
+
+    let balance = geth_erc1155_balance(maker_address, U256::from(token_id));
+    assert_eq!(U256::from(amount), balance);
+}
+
+#[test]
+fn send_and_refund_erc721_maker_payment_secret() {
+    thread::sleep(Duration::from_secs(5));
+    let token_id = 3u32;
+    let time_lock_to_refund = now_sec() + 1000;
+
+    let setup = setup_test(
+        token_id,
+        None,
+        ContractType::Erc721,
+        geth_nft_maker_swap_v2(),
+        geth_erc721_contract(),
+        time_lock_to_refund,
+    );
+
+    let maker_payment_to_refund = send_nft_maker_payment(&setup, 1.into());
+    log!(
+        "Maker sent ERC721 NFT payment, tx hash: {:02x}",
+        maker_payment_to_refund.tx_hash()
+    );
+
+    wait_for_confirmations(&setup.maker_global_nft, &maker_payment_to_refund, 150);
+    let current_owner = geth_erc712_owner(U256::from(token_id));
+    assert_eq!(current_owner, geth_nft_maker_swap_v2());
+
+    let refund_secret_tx = refund_nft_maker_payment(
+        &setup,
+        &maker_payment_to_refund,
+        &ContractType::Erc721,
+        RefundType::Secret,
+    );
+    log!(
+        "Maker refunded ERC721 NFT payment using Taker secret, tx hash: {:02x}",
+        refund_secret_tx.tx_hash()
+    );
+
+    wait_for_confirmations(&setup.maker_global_nft, &refund_secret_tx, 150);
+    let current_owner = geth_erc712_owner(U256::from(token_id));
+    let maker_address = block_on(setup.maker_global_nft.my_addr());
+    assert_eq!(current_owner, maker_address);
+}
+
+#[test]
+fn send_and_refund_erc1155_maker_payment_secret() {
+    thread::sleep(Duration::from_secs(2));
```

Comment:
Why do we need this sleep at the start of the test? just curious :)

Reply:
I had situations where some tests were failing due to `confirmation failed` issue. Even after rerunning docker tests, the problem persisted. Adjusting the sleep duration helped reduce the frequency of these confirmation issues.

Reply:
> I had situations where some tests were failing due to confirmation failed issue

Can you show me logs related to this issue, it might be a problem that can be solved. Nonce issues are the usual suspects in these situations.

Reply:
> Can you show me logs related to this issue, it might be a problem that can be solved. Nonce issues are the usual suspects in these situations.

Found logs from my notes
```
 ---- docker_tests::eth_docker_tests::send_and_spend_erc1155_maker_payment stdout ----
· 2024-04-18 11:05:47 +0000 [sign] get_gas_price…
· 2024-04-18 11:05:47 +0000 [sign-and-send] get_addr_nonce…
18 11:05:47, eth_docker_tests:805] Maker sent ERC1155 NFT Payment tx hash a3524dd8929af757817dfa0450a8e21e63b691da1d7da41403ebdb6bf5cc52af
· 2024-04-18 11:05:47 +0000 [NFT_ETH] Waiting for confirmations… Failed.
thread 'docker_tests::eth_docker_tests::send_and_spend_erc1155_maker_payment' panicked at 'called `Result::unwrap()` on an `Err` value: "eth:5002] Internal: eth:5002] Tx receipt Receipt { transaction_hash: 0xa3524dd8929af757817dfa0450a8e21e63b691da1d7da41403ebdb6bf5cc52af, transaction_index: 0, block_hash: Some(0x258427083fd1fbcf0231f81e62a620d7e69e6753f221458e457ab77b0217ae28), block_number: Some(92), from: 0x0ee824ffa08f316065eaf7209d2e5125a520495b, to: Some(0xfb00e3aca00f988d70ca683a3b04a3e08d095b50), cumulative_gas_used: 148971, gas_used: Some(148971), contract_address: None, logs: [], status: Some(0), root: None, logs_bloom: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, transaction_type: Some(0), effective_gas_price: Some(1000007365) } status of NFT_ETH tx 0xa3524dd8929af757817dfa0450a8e21e63b691da1d7da41403ebdb6bf5cc52af is failed"', mm2src/mm2_main/tests/docker_tests/eth_docker_tests.rs:814:67
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- docker_tests::eth_docker_tests::send_and_spend_erc721_maker_payment stdout ----
· 2024-04-18 11:05:48 +0000 [sign] get_gas_price…
· 2024-04-18 11:05:48 +0000 [sign-and-send] get_addr_nonce…
18 11:05:48, eth_docker_tests:724] Maker sent ERC721 NFT Payment tx hash 16121294842ebda1818a3311e6cc2de304843fce09f3d026105a74d7feb76c22
· 2024-04-18 11:05:48 +0000 [NFT_ETH] Waiting for confirmations… Failed.
thread 'docker_tests::eth_docker_tests::send_and_spend_erc721_maker_payment' panicked at 'called `Result::unwrap()` on an `Err` value: "eth:5002] Internal: eth:5002] Tx receipt Receipt { transaction_hash: 0x16121294842ebda1818a3311e6cc2de304843fce09f3d026105a74d7feb76c22, transaction_index: 0, block_hash: Some(0x01ee88be835604ae24c0a4ca5ea5d3c342fa6b39ef12c10ade4158fbe8bcfad2), block_number: Some(103), from: 0x69f1da705635c49ef11dd7138a6f7f08c96b0e34, to: Some(0xdee6088c0d3b49fc06d5f46a639a38c5d9e0de49), cumulative_gas_used: 149054, gas_used: Some(149054), contract_address: None, logs: [], status: Some(0), root: None, logs_bloom: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, transaction_type: Some(0), effective_gas_price: Some(1000002638) } status of NFT_ETH tx 0x16121294842ebda1818a3311e6cc2de304843fce09f3d026105a74d7feb76c22 is failed"', mm2src/mm2_main/tests/docker_tests/eth_docker_tests.rs:733:67
```

Thats pretty strange, In geth tests we generate random privkey each time, so taker/maker should be unique and the problem of sending multiple transactions at the same time from one wallet shouldn't occur.

Reply:
ooh, it's most probably that the transaction used all gas due to using v1 gas limits
https://github.com/KomodoPlatform/komodo-defi-framework/blob/c58808a35d8c9565b067ce9eb10a4dce8b0d205d/mm2src/coins/eth.rs#L240
In the transactions you posted you can see that `gas_used: Some(148971)` and `gas_used: Some(149054)`. The remaining gas can't carry out the next instruction and it fails. Try increasing gas limits and see if these issues are resolved, if they are resolved, then please set some new limits for v2 swaps.
Related: https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129#discussion_r1685165050

Reply:
Hmm agree, in the case above its probably due to gas limit 150_000.

Right now I dont have any issues with erc1155/erc720 transactions https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/10144458950/job/28047990228?pr=2169

I also found these logs in my notes:

 `gas_used: Some(30839)`
```
---- docker_tests::eth_docker_tests::send_send_and_refund_erc721_maker_payment_timelock stdout ----

All pending transactions have been confirmed.

27 14:00:35, eth_docker_tests:1189] time_lock_to_refund: 1716819437

27 14:00:38, eth_docker_tests:1214] Maker sent ERC721 NFT payment, tx hash: 28cd786a40707ea92bb10d8d690d07ae87d72c282e3dd426196038dbfb23ff05

· 2024-05-27 14:00:49 +0000 [NFT_ETH] Waiting for confirmations… Confirmed.

All pending transactions have been confirmed.

htlc_params: [FixedBytes([3, 103, 242, 248, 247, 139, 180, 216, 242, 182, 38, 126, 32, 239, 123, 84, 103, 225, 131, 49, 85, 70, 4, 158, 42, 136, 68, 50, 48, 189, 11, 49]), Address(0x16e281a9f2e7581269a13e516aa79d6a4a1cd980), Address(0xbac1c9f2087f39caaa4e93412c6412809186870e), FixedBytes([117, 135, 123, 180, 29, 57, 59, 95, 184, 69, 92, 230, 14, 205, 141, 218, 0, 29, 6, 49, 100, 150, 177, 77, 250, 127, 137, 86, 86, 238, 202, 74]), FixedBytes([114, 205, 110, 132, 34, 196, 7, 251, 109, 9, 134, 144, 241, 19, 11, 125, 237, 126, 194, 247, 245, 225, 211, 11, 217, 213, 33, 240, 21, 54, 55, 147]), Uint(1716819437)]

args.taker_secret_hash: [117, 135, 123, 180, 29, 57, 59, 95, 184, 69, 92, 230, 14, 205, 141, 218, 0, 29, 6, 49, 100, 150, 177, 77, 250, 127, 137, 86, 86, 238, 202, 74]

args.maker_secret_hash: [114, 205, 110, 132, 34, 196, 7, 251, 109, 9, 134, 144, 241, 19, 11, 125, 237, 126, 194, 247, 245, 225, 211, 11, 217, 213, 33, 240, 21, 54, 55, 147]

decoded: [Address(0xe9f3aaeb5a90f230a4ce78be737bb6d678f9baac), Address(0x9eb88cd58605d8fb9b14652d6152727f7e95fb4d), Uint(2), Bytes([3, 103, 242, 248, 247, 139, 180, 216, 242, 182, 38, 126, 32, 239, 123, 84, 103, 225, 131, 49, 85, 70, 4, 158, 42, 136, 68, 50, 48, 189, 11, 49, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 22, 226, 129, 169, 242, 231, 88, 18, 105, 161, 62, 81, 106, 167, 157, 106, 74, 28, 217, 128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 186, 193, 201, 242, 8, 127, 57, 202, 170, 78, 147, 65, 44, 100, 18, 128, 145, 134, 135, 14, 117, 135, 123, 180, 29, 57, 59, 95, 184, 69, 92, 230, 14, 205, 141, 218, 0, 29, 6, 49, 100, 150, 177, 77, 250, 127, 137, 86, 86, 238, 202, 74, 114, 205, 110, 132, 34, 196, 7, 251, 109, 9, 134, 144, 241, 19, 11, 125, 237, 126, 194, 247, 245, 225, 211, 11, 217, 213, 33, 240, 21, 54, 55, 147, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 102, 84, 149, 237])]

27 14:00:56, eth_docker_tests:1241] Maker refunded ERC721 NFT Maker payment Timelock, tx hash: d5762cca88997d46d38743648bd148d8cba53f0378a2aca5a974d4ba91ff911c

· 2024-05-27 14:01:00 +0000 [NFT_ETH] Waiting for confirmations… Failed.

thread 'docker_tests::eth_docker_tests::send_send_and_refund_erc721_maker_payment_timelock' panicked at 'called `Result::unwrap()` on an `Err` value: "eth:5248] Internal: eth:5248] Tx receipt Receipt { transaction_hash: 0xd5762cca88997d46d38743648bd148d8cba53f0378a2aca5a974d4ba91ff911c, transaction_index: 13, block_hash: Some(0xd9559f27bbd0a0a41f88a3061dd61e2ee3654ac1d5c71a35b3d4c35c1f349e08), block_number: Some(5988387), from: 0x16e281a9f2e7581269a13e516aa79d6a4a1cd980, to: Some(0x9eb88cd58605d8fb9b14652d6152727f7e95fb4d), cumulative_gas_used: 861654, gas_used: Some(30839), contract_address: None, logs: [], status: Some(0), root: None, logs_bloom: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, transaction_type: Some(0), effective_gas_price: Some(69699307768) } status of NFT_ETH tx 0xd5762cca88997d46d38743648bd148d8cba53f0378a2aca5a974d4ba91ff911c is failed"', mm2src/mm2_main/tests/docker_tests/eth_docker_tests.rs:1252:67

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```


`gas_used: Some(73001)`
```
---- docker_tests::nft_swap_proto_v2_tests::send_and_spend_erc721_maker_payment stdout ----
· 2024-03-20 10:16:17 +0000 [sign] get_gas_price…
· 2024-03-20 10:16:17 +0000 [sign-and-send] get_addr_nonce…
Maker sent ERC721 NFT Payment tx hash 30ca863495e48a814c56c98c484b1c547afadde3fcefe5a2d4379be3100cb365
· 2024-03-20 10:16:17 +0000 [NFT_ETH] Waiting for confirmations… Failed.
thread 'docker_tests::nft_swap_proto_v2_tests::send_and_spend_erc721_maker_payment' panicked at 'called `Result::unwrap()` on an `Err` value: "eth:4995] Internal: eth:4995] Tx receipt Receipt { transaction_hash: 0x30ca863495e48a814c56c98c484b1c547afadde3fcefe5a2d4379be3100cb365, transaction_index: 0, block_hash: Some(0xda206cad847f1b11ba23ed4e404663d591367c42998b4143fc8f6898cc692958), block_number: Some(35), from: 0xdb3e6328007939e62ab9f8d49d4f40ef0eb998b7, to: Some(0x298afdfe93297ad2d3114d18869125e1e7e1f212), cumulative_gas_used: 73001, gas_used: Some(73001), contract_address: None, logs: [], status: Some(0), root: None, logs_bloom: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, transaction_type: Some(0), effective_gas_price: Some(1014823722) } status of NFT_ETH tx 0x30ca863495e48a814c56c98c484b1c547afadde3fcefe5a2d4379be3100cb365 is failed"', mm2src/mm2_main/tests/docker_tests/nft_swap_proto_v2_tests.rs:49:67
```

I wold like to update `transaction_confirmed_at` function and return not only block, but also `gas_used` from receipt,
https://github.com/KomodoPlatform/komodo-defi-framework/blob/8a16714ef2a314ef9ca3385dcaea63a8b1feffec/mm2src/coins/eth.rs#L5258

and include gas used value in `Confirmed` status log
https://github.com/KomodoPlatform/komodo-defi-framework/blob/8a16714ef2a314ef9ca3385dcaea63a8b1feffec/mm2src/coins/eth.rs#L2325

In the logs, we currently see the gas used for failed transactions. It would be useful to also include the gas used for previous successful transactions. I would like to add this in the next PR.
Regarding gas limits, I can remove all sleep durations from the  eth docker tests and observe the results.

ps: I will start saving logs of failed confirmations in my notes to ensure they are not lost when CI results expire.
ps2: also had moments of falinig legacy swaps, e.g. 
![Screenshot 2024-06-05 at 10 01 42](https://github.com/user-attachments/assets/8f0ef321-2d91-42fa-b00e-43b016544ed3)

![Screenshot 2024-06-10 at 17 02 57](https://github.com/user-attachments/assets/64c94939-b707-4509-a4a1-a28cedf8c7e8)

Reply:
> Regarding gas limits, I can do such experiment - remove all sleep durations from the eth docker tests and observe the results.

I wanted the sleep durations to be removed in this PR, but if not, please don't forget about it as I will approve and merge the PR after your confirmation that we will remove these sleeps in the other PR.

Reply:
> I wanted the sleep durations to be removed in this PR, but if not, please don't forget about it as I will approve and merge the PR after your confirmation that we will remove these sleeps in the other PR.

Lest remove duration sleep in this PR https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169 as it will be r2r at the end of week (and has working docker tests) and next week is review week.

Reply:
> I wanted the sleep durations to be removed in this PR, but if not, please don't forget about it as I will approve and merge the PR after your confirmation that we will remove these sleeps in the other PR.

https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/95f21dde82e790b6a7d2cd97f5f6d2558944d0b4
https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/10194809417/job/28202160615?pr=2169
removed sleep duration in eth docker tests (with exception of sepolia). 
I can confirm that current Geth tests pass. I suppose geth tests are unstable and confirmation can fail sometimes due to geth block generation or other geth reasons.